File: 00704-binary-search/solution.py
from typing import List

class Solution:
    def search(self, nums: List[int], target: int) -> int:
        low = 0
        high = len(nums) - 1

        while low <= high:
            med = (low + high) // 2
            if target == nums[med]:
                return med
            elif target < nums[med]:
                high = med - 1
            else:
                low = med + 1

        return -1

File: 00704-binary-search/solution.cs
public class Solution {
    public int Search(int[] nums, int target) {
        int low = 0;
        int high = nums.Length - 1;

        while (low <= high) {
            int med = (low + high) / 2;

            if (nums[med] == target) {
                return med;
            } else if (target < nums[med]) {
                high = med - 1;
            } else { // target > nums[med]
                low = med + 1;
            }
        }

        return -1;
    }
}
